test/integration/test_request_source_plugin.py

```diff
@@ -0,0 +1,62 @@
+"""Tests for the nighthawk_service binary."""
+
+import pytest
+import os
+
+from test.integration.integration_test_fixtures import (http_test_server_fixture, server_config)
+from test.integration import utility
+from test.integration import asserts
+
+
+def test_request_source_plugin_happy_flow(http_test_server_fixture):
```

Comment:
Just a thought, I wonder if it would make sense to parameterize this test over plugin setup. So we iterate the same test over all known plugins. That way, future plugins could add themselves to this hypothetical parameterized list, and have basic integration test coverage through that. (This would imply that all plugin types would be set up with a configuration that yields the same traffic pattern). Wouldn't object to leaving this as is, but if you see sufficient value it might be worth doing.

Reply:
Sure thing.